pydantic/generics.py

```diff
@@ -132,16 +187,16 @@ def _build_generic_fields(
 
 
 def _parameterize_generic_field(field_type: Type[Any], typevars_map: Dict[TypeVarType, Type[Any]]) -> Type[Any]:
-    if lenient_issubclass(field_type, GenericModel) and not field_type.__concrete__:
+    try:
+        is_model = isinstance(field_type, GenericModel) and not field_type.__concrete__
```

Comment:
I'm not clear why this changed is required?

Reply:
Unfortunately `lenient_issubclass` fails if `field_type` is a generic alias.

```python
lenient_issubclass(list[int], GenericModel)
```
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/dek/Documents/Code/pydantic/pydantic/utils.py", line 145, in lenient_issubclass
    return isinstance(cls, type) and issubclass(cls, class_or_tuple)
  File "/home/dek/anaconda/envs/pydantic-39/lib/python3.9/abc.py", line 102, in __subclasscheck__
    return _abc_subclasscheck(cls, subclass)
TypeError: issubclass() arg 1 must be a class
```

The exact logic is that:
```
isinstance(list[int], type)
```
returns `True` but then `issubclass` needs a `class` which `list[int]` is not.

I can add workaround code to lenient_issubclass but I wasn't sure that would be the right thing to do given how many places lenient_issubclass is used and this sort of handling of typehints seemed relatively specialized towards generics.